core/src/main/java/ai/timefold/solver/core/impl/domain/variable/cascade/AbstractCascadingUpdateShadowVariableListener.java

```diff
@@ -6,39 +6,51 @@
 import ai.timefold.solver.core.api.domain.variable.VariableListener;
 import ai.timefold.solver.core.api.score.director.ScoreDirector;
 import ai.timefold.solver.core.impl.domain.common.accessor.MemberAccessor;
+import ai.timefold.solver.core.impl.domain.variable.cascade.command.CascadingUpdateCommand;
+import ai.timefold.solver.core.impl.domain.variable.cascade.command.Pair;
+import ai.timefold.solver.core.impl.domain.variable.descriptor.ListVariableDescriptor;
 import ai.timefold.solver.core.impl.domain.variable.descriptor.VariableDescriptor;
 
 /**
  * @param <Solution_> the solution type, the class with the {@link PlanningSolution} annotation
  */
 public abstract class AbstractCascadingUpdateShadowVariableListener<Solution_> implements VariableListener<Solution_, Object> {
 
+    private final CascadingUpdateCommand<Pair<Integer, Object>> indexElementCommand;
+    final ListVariableDescriptor<Solution_> sourceListVariableDescriptor;
     final List<VariableDescriptor<Solution_>> targetVariableDescriptorList;
     final MemberAccessor targetMethod;
```

Comment:
`protected` would increase visibility; `package-private` (the default when no modifier is used) has lower visibility than `protected`. +1 to making it private. (In general, do not use `package-private`; always prefer either `protected` or `private`.

Reply:
(`protected` when subclasses need to access the field/override a method, private otherwise)

Reply:
It's true that `protected` would _increase_ visibility, but it would only expose internals to its children, which - in my mind - actually _decreases_ visibility. Since no one from outside the hierarchy gets to see.

But yes, if `private` is possible, that's the best choice.